src/main/java/sysc4806/graduateAdmissions/service/TermManagerImpl.java

```diff
@@ -0,0 +1,43 @@
+package sysc4806.graduateAdmissions.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.validation.annotation.Validated;
+import sysc4806.graduateAdmissions.dto.TermDTO;
+import sysc4806.graduateAdmissions.mapper.TermMapper;
+import sysc4806.graduateAdmissions.model.Term;
+import sysc4806.graduateAdmissions.repositories.TermRepository;
+
+import javax.validation.Valid;
+import java.util.Optional;
+
+@Service
+@Validated
+public class TermManagerImpl implements TermManager {
+
+    @Autowired
+    private TermRepository termRepository;
+    @Autowired
+    private TermMapper termMapper;
+
+    @Override
+    public Optional<TermDTO> createTerm(@Valid TermDTO termDTO) {
+        if (termDTO.getTermId() != null && termRepository.existsById(termDTO.getTermId())) {
+            return Optional.empty();
+        }
+        Term newTerm = termMapper.map(termDTO);
+        return Optional.of(termMapper.map(termRepository.save(newTerm)));
+    }
+
+    @Override
+    public Optional<TermDTO> updateTerm(@Valid TermDTO termDTO) {
+        Optional<Term> existingTerm = termRepository.findById(termDTO.getTermId());
+        if (existingTerm.isPresent()) {
+            Term updatedTerm = termRepository.save(termMapper.map(termDTO, existingTerm.get()));
+            return Optional.ofNullable(termMapper.map(updatedTerm));
+        } else {
+            return Optional.empty();
+        }
+    }
```

Comment:
You might ask, why do we need DTOs and this service layer?

If we skip the service layer, our api is tightly coupled with the resource (table's) data, meaning if I wanted to add a column in the User entity, say a password, but I got this password from a third party password management service and not our endpoint, by using entities in the endpoints, we're now exposing this property in our api when we don't want it there.
With the DTO, we can specify exactly what we want to get from the user, send that to the service layer, then the service layer will have all the logic in it to handle grabbing all it needs to persist the info, including grabbing that password from another place.
We do a lot of DTO <-> Entity conversions, but c'est la vie. I've tried many other methods and all of them have their own issues.
Note that GET and DELETE endpoints tend to not need the service layer since they don't take json payloads and therefore, the entity/our db schema isn't exposed.

Reply:
this makes sense, but just as a reminder, third party login means we don't need password field in user anymore. The 3rd party does all the password managing, and we just get a token from the user that they got from google saying that they have a valid account.

I think you probably just picked a field at random for your explanation, but just in case (and so this is written somewhere for everyone to be aware of)